presto-docs/src/main/sphinx/functions/lambda.rst

```diff
@@ -0,0 +1,67 @@
+============================
+Lambdas and Lambda Functions
+============================
+
+Lambda Expression
+-----------------
+
+Lambda expressions in Presto are written with ``->``. For example::
+
+    x -> x + 1
+    (x, y) -> x + y
+    x -> regexp_like(x, 'a+')
+    x -> x[1] / x[2]
+    x -> IF(x > 0, x, -x)
+    x -> COALESCE(x, 0)
+    x -> CAST(x AS JSON)
+
+Most expressions you can write in Presto SQL can appear in lambda body, with a few exceptions:
+
+* ``TRY`` function is not supported yet. (:func:`TRY_CAST` is supported.)
+* Capture is not supported yet. In other words,
+
+  * Columns or relations can not be referenced.
+  * Only lambda variables from the inner-most lambda expression can be referenced.
+
+* Subqueries are not supported.
+
+Lambda Functions
+----------------
+
+.. function:: transform(array ARRAY(T), function FUNCTION(T,U)) -> ARRAY(U)
+
+    Returns an array that applies ``function`` to each element of ``array``. For example::
+
+        transform(ARRAY [], x -> x + 1) -> ARRAY []
```

Comment:
Make all these `SELECT` queries with the output at end with a comment. Using an arrow is really confusing given that it's also part of the lambda syntax which is being demonstrated. See JSON functions for examples.